Mathlib/Data/Matroid/IndepAxioms.lean

```diff
@@ -0,0 +1,559 @@
+/-
+Copyright (c) 2023 Peter Nelson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Peter Nelson
+-/
+import Mathlib.Data.Matroid.Basic
+
+/-!
+Matroids in mathlib are defined axiomatically in terms of bases,
+but can be described just as naturally via their collections of independent sets,
+and in fact such a description, being more 'verbose', can often be useful.
+As well as this, the definition of a `Matroid` uses an unwieldy 'maximality'
+axiom that can be dropped in cases where there is some finiteness assumption.
+
+This file provides several ways to do define a matroid in terms of its independence or base
+predicates, using axiom sets that are appropriate in different settings,
+and often much simpler than the general definition.
+It also contains `simp` lemmas and typeclasses as appropriate.
+
+All the independence axiom sets need nontriviality (the empty set is independent),
+monotonicity (subsets of independent sets are independent),
+and some form of 'augmentation' axiom, which allows one to enlarge a non-maximal independent set.
+This augmentation axiom is still required when there are finiteness assumptions, but is simpler.
+It just states that if `I` is a finite independent set and `J` is a larger finite
+independent set, then there exists `e ∈ J \ I` for which `insert e I` is independent.
+This is the axiom that appears in all most of the definitions.
+
+## Main definitions
+
+* `matroid_of_base` constructs a matroid from the base axioms (using the definition of a matroid).
```

Comment:
```suggestion
* `Matroid.matroid_of_base` constructs a matroid from the base axioms (using the definition of a matroid).
```
and throughout, so that these generate doc links